NDTensors/src/UnallocatedArrays/src/unallocated_impl.jl

```diff
@@ -0,0 +1,29 @@
+for Typ in (:UnallocatedFill, :UnallocatedZeros)
+  ## Here are functions specifically defined for UnallocatedArrays
+  ## not implemented by FillArrays
+  @eval begin
+    alloctype(A::$Typ) = alloctype(typeof(A))
+    alloctype(::Type{<:$Typ{ElT,N,Axes,Alloc}}) where {ElT,N,Axes,Alloc} = Alloc
+
+    get_index(A::$Typ) = getindex(data(A))
+
+    array(A::$Typ) = alloctype(typeof(A))(data(A))
+    Array(A::$Typ) = array(A)
+    ## A function for NDTensors to launch functions off of
+    is_immutable(A::$Typ) = is_immutable(typeof(A))
+    is_immutable(::Type{<:$Typ}) = true
```

Comment:
Could you explain what this function is for?

Reply:
In the PR I made to replace `EmptyStorage` I think I used this
```
## Check datatypes to see if underlying storage is a 
## UnallocatedZeros
is_unallocated_zeros(a) = data_isa(a, UnallocatedZeros)
```
so I wasn't sure if I would need functionality like to to say thaw the immutable data type into a mutable version. I am not sure its really necessary because in that PR I also had the `data_isa` which could be used for the same purpose. I can remove